### PR TITLE
Implement data process graph feature

### PR DESCRIPTION
**Pull request by liangjun-jiang**
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable)

1. the expected result
Screen Shot 2020-06-09 at 2 54 17 PM

2.  has been updated to give more details about onboarding the graph feature

### CODE REVIEW

**Comment by jywadhwani** [comment]
Let's also add test for graph builder similar to `DatasetGraphBuilderTest`?

File: metadata-dao-impl/neo4j-dao/src/main/java/com/linkedin/metadata/dao/Neo4jUtil.java [hunk 1]
```diff
@@ -44,6 +46,7 @@ private Neo4jUtil() {
     {
       put(CorpuserUrn.ENTITY_TYPE, getType(CorpUserEntity.class));
       put(CorpGroupUrn.ENTITY_TYPE, getType(CorpGroupEntity.class));
+      put(DataProcessUrn.ENTITY_TYPE, getType(DataProcessEntity.class));
```
Comment (keremsahin1) [inline metadata-dao-impl/neo4j-dao/src/main/java/com/linkedin/metadata/dao/Neo4jUtil.java:49]: This map was a quick hack :) We're getting rid of this with this PR: https://github.com/linkedin/datahub/pull/1700

**Comment by liangjun-jiang** [comment]
> Let's also add test for graph builder similar to `DatasetGraphBuilderTest`?

unit test added.

File: metadata-events/mxe-schemas/src/original/avro/com/linkedin/common/Deprecation.avsc [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{
```
Comment (keremsahin1) [inline metadata-events/mxe-schemas/src/original/avro/com/linkedin/common/Deprecation.avsc:1]: Can you get rid of these Avro files? They are auto-generated.
`rm -rf metadata-events/mxe-schemas/src/original`

File: metadata-builders/src/main/java/com/linkedin/metadata/builders/graph/DataProcessGraphBuilder.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  @Nonnull
+  @Override
+  protected List<? extends RecordTemplate> buildEntities(@Nonnull DataProcessSnapshot snapshot) {
+    final DataProcessUrn urn = snapshot.getUrn();
+    final DataProcessEntity entity = new DataProcessEntity().setUrn(urn)
+        .setName(urn.getNameEntity())
+        .setOrchestrator(urn.getOrchestrator())
+        .setOrigin(urn.getOriginEntity());
+
+    setRemovedProperty(snapshot, entity);
```
Comment (jywadhwani) [inline metadata-builders/src/main/java/com/linkedin/metadata/builders/graph/DataProcessGraphBuilder.java:38]: Do we really need this? `DataProcess` doesn't seem to have the `Status` aspect.

**Comment by liangjun-jiang** [comment]
> Can you also please update the part of the image for `gma graph`? I think we don't need step 3. I'm also not sure about step 4?

You are correct. We don't need step 3 & 4. Our internal version of Datahub is not synced with this open source one, so 3 & 4 are still needed.

**Pull request merged**